### PR TITLE
[5.3] PHPStan: Fixing return statements and types

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
PHPStan is reporting a bunch of errors where return types are not set correctly. This PR fixes all 25 occurences of this.

### Testing Instructions
Codereview.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_templates/src/Model/TemplateModel.php [hunk 1]
```diff
@@ -306,13 +306,14 @@ public function prepareCoreFiles($dir, $element, $template)
...
      *
-     * @return  integer  Number of files changed.
+     * @return  bool|\RuntimeException  Update successfull or exception object
      *
      * @since   4.0.0
      */
     public function publish($ids, $value, $exid)
     {
         $db = $this->getDatabase();
+        $result = false;
```
Comment (richard67) [inline administrator/components/com_templates/src/Model/TemplateModel.php]:
```suggestion
        $db     = $this->getDatabase();
        $result = false;
```
Fix PHPCS failing in Drone.

File: plugins/user/terms/src/Extension/Terms.php [hunk 2]
```diff
@@ -60,6 +60,8 @@ public function onContentPrepareForm(Form $form, $data)
         // Push the terms and conditions article ID into the terms field.
         $form->setFieldAttribute('terms', 'article', $termsarticle, 'terms');
         $form->setFieldAttribute('terms', 'note', $termsnote, 'terms');
+
+        return true;
```
Comment (Quy) [inline plugins/user/terms/src/Extension/Terms.php:64]: Change to `@return  true`?

File: administrator/components/com_templates/src/Model/TemplateModel.php [hunk 3]
```diff
@@ -1620,13 +1639,17 @@ public function getPreview()
             $result = $db->loadObject();
         } catch (\RuntimeException $e) {
             $app->enqueueMessage($e->getMessage(), 'warning');
+
+            return null;
```
Comment (HLeithner) [inline administrator/components/com_templates/src/Model/TemplateModel.php:1667]: doc block says we only return `object` that needs to be changed

Comment (laoneo) [reply administrator/components/com_templates/src/Model/TemplateModel.php:1667]: @Hackwar can you have a look here?

Comment (Hackwar) [reply administrator/components/com_templates/src/Model/TemplateModel.php:1667]: Done.

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**